deb-drop.go

```diff
@@ -642,16 +642,31 @@ func (h *handler) batchPkgs(pkgs []string, nPrefix int) [][]string {
 }
 
 func (h *handler) addToRepos(content io.Reader, repos []string, packageName string) error {
+
+	// First ensure tmpDir from config exists
+	stat, err := os.Stat(h.config.TmpDir)
+	if err != nil {
```

Comment:
you could also combine it this way to have `err` only in this scope and avoid second err check:
```go
if stat, err := os.Stat(h.config.TmpDir); err != nil {
	if os.IsNotExist(err) {
		logrus.Infof("Tmp directory %s does not exist. Creating..", h.config.TmpDir)
		err = os.Mkdir(h.config.TmpDir, os.ModePerm)
	} else if !stat.IsDir() {
		err = fmt.Errorf("tmp location %s exists, but it is not a directory", h.config.TmpDir)
	}

	if err != nil {
		return err
	}
}
```

Reply:
but then it will still never reach the !stat.IsDir() as err is nil, maybe something like:
```go
// First ensure tmpDir from config exists
if stat, err := os.Stat(h.config.TmpDir); err != nil {
	if os.IsNotExist(err) {
		logrus.Infof("Tmp directory %s does not exist. Creating..", h.config.TmpDir)
		if err = os.Mkdir(h.config.TmpDir, os.ModePerm); err != nil {
			return err
		}
	} else {
		return fmt.Errorf("failed to check tmp location %s: %w", h.config.TmpDir, err)
	}
} else if !stat.IsDir() {
	return fmt.Errorf("tmp location %s exists, but it is not a directory", h.config.TmpDir)
}
```